lib/config/migration.ts

```diff
@@ -110,6 +80,14 @@ export function migrateConfig(config: RenovateConfig): MigratedConfig {
           for (const item of migratedConfig[key] as unknown[]) {
             if (is.object(item) && !is.array(item)) {
               const arrMigrate = migrateConfig(item as RenovateConfig);
+              if (
+                (item as PackageRule).matchDepTypes?.some((depType) =>
+                  depTypes.includes(depType)
+                )
+              ) {
+                (arrMigrate as unknown as PackageRule).matchDepTypes = [key];
+                delete (arrMigrate as unknown as PackageRule).packageRules;
+              }
```

Comment:
Can you explain why you needed to add this part? Normally we're removing

Reply:
This is a part of the old logic which when left in the class definition of devDependencies, engines, ... caused import issues. So I added this here.